app/controllers/discourse_assign/assign_controller.rb

```diff
@@ -44,11 +44,27 @@ def unassign
     end
 
     def assign
-      reassign_or_assign_target(action: "assign")
-    end
+      target_id = params.require(:target_id)
+      target_type = params.require(:target_type)
+      username = params.permit(:username)['username']
+      group_name = params.permit(:group_name)['group_name']
 
-    def reassign
-      reassign_or_assign_target(action: "reassign")
+      assign_to = username.present? ? User.find_by(username_lower: username.downcase) : Group.where("LOWER(name) = ?", group_name.downcase).first
+
+      raise Discourse::NotFound unless assign_to
+      raise Discourse::NotFound if !Assignment.valid_type?(target_type)
+      target = target_type.constantize.where(id: target_id).first
+      raise Discourse::NotFound unless target
+
+      # perhaps?
+      #Scheduler::Defer.later "assign topic" do
```

Comment:
Is the `assign` method a CPU/DB intensive process?

Reply:
From looking at the logs on assign, it does not seem overly heavy... this was a comment (left over) from before I started working on the re-assign workflow, so i figured I would leave it in. In the case someone was coming back to update? 🤷‍♂️

Reply:
I'd rather we remove the comment to keep the code clean (it'll be in the git history anyways). We can always revisit this later if we end up hitting a performance wall.